flink-end-to-end-tests/test-scripts/test_ha_datastream.sh

```diff
@@ -100,4 +102,21 @@ STATE_BACKEND_FILE_ASYNC=${2:-true}
 STATE_BACKEND_ROCKS_INCREMENTAL=${3:-false}
 ZOOKEEPER_VERSION=${4:-3.4}
 
-run_ha_test 4 ${STATE_BACKEND_TYPE} ${STATE_BACKEND_FILE_ASYNC} ${STATE_BACKEND_ROCKS_INCREMENTAL} ${ZOOKEEPER_VERSION}
+function kill_test_watchdog() {
+    local watchdog_pid=`cat $TEST_DATA_DIR/job_watchdog.pid`
+    echo "Stopping job timeout watchdog (with pid=$watchdog_pid)"
+    kill $watchdog_pid
+}
+on_exit kill_test_watchdog
+
+( 
+    cmdpid=$BASHPID; 
```

Comment:
`BASHPID` does not work on my Mac. It's introduced in Bash 4, however the version of Bash on my Mac is 3.2. Maybe we could use `$$` when `BASHPID` is not supported? `$$` is more general, however sometimes it does not work well as expected, like executing in background.

Reply:
Yes, these changes require bash 4. Bash on mac is pretty outdated sadly.

I think `$$` won't work, as it will return the PID of the bash-process, not of the subshell I'm launching there:

> When you want to find out the PID of your current shell session, even from subshells, that’s when you use $$. Any subshell has its own PID as well, and you can access it via the $BASHPID variable.

https://bashwizard.com/shells-subshells-pids/

But I will add a warning to the script, that is requires at least bash 4

Reply:
Ah, you are right. It's in another sub-process.

I come up with a work-around way. It probably works that executing the watchdog in original main process and executing the testing scripts in sub-process. We can get the sub-process pid in main process.

But I guess it's not a big deal, as MacOS is not a main scenario. I hope this PR could be merged asap. I would help a lot.